process/notifier/sovereignNotifier.go

```diff
@@ -126,77 +135,50 @@ func checkNilOutportBlockFields(outportBlock *outport.OutportBlock) error {
 	return nil
 }
 
-func (notifier *sovereignNotifier) createAllIncomingMbs(txPool *outport.TransactionPool) ([]*block.MiniBlock, error) {
-	mbs := make([]*block.MiniBlock, 0)
+func (notifier *sovereignNotifier) createIncomingEvents(logsData []*outport.LogData) []*transaction.Event {
+	incomingEvents := make([]*transaction.Event, 0)
 
-	txsMb, err := notifier.createIncomingMbsFromTxs(txPool.Transactions)
-	if err != nil {
-		return nil, err
+	for _, logData := range logsData {
+		events := logData.GetLog().Events
+		eventsFromLog := notifier.getIncomingEvents(events, logData.TxHash)
+
+		incomingEvents = append(incomingEvents, eventsFromLog...)
 	}
 
-	// TODO: when specs are defined, we should also handle scrs mbs
-	mbs = append(mbs, txsMb...)
-	return mbs, nil
+	return incomingEvents
 }
 
-func (notifier *sovereignNotifier) createIncomingMbsFromTxs(txs map[string]*outport.TxInfo) ([]*block.MiniBlock, error) {
-	execOrderTxHashMap := make(map[string]uint32)
-	shardIDTxHashMap := make(map[uint32][][]byte)
+func (notifier *sovereignNotifier) getIncomingEvents(events []*transaction.Event, txHash string) []*transaction.Event {
+	incomingEvents := make([]*transaction.Event, 0)
 
-	for txHash, tx := range txs {
-		receiver := tx.GetTransaction().GetRcvAddr()
-		_, found := notifier.subscribedAddresses[string(receiver)]
-		if !found {
+	for _, event := range events {
+		if !notifier.isSubscribed(event, txHash) {
```

Comment:
Yes, and I would keep it as param for logging purposes, helps a lot.
Changed `getIncomingEvents` to receive a `logData` as param, but couldn't change `isSubscribed` as well, since
- `getIncomingEvents` needs an array or events
- `isSubscribed` needs only one event

Reply:
ok